lib/pages/timetable/service/uni_event_provider.dart

```diff
@@ -297,6 +306,156 @@ class UniEventProvider extends EventProvider<UniEventInstance>
+
+      newRruleString = newRruleString.replaceAll(RegExp(r'DAILY'), 'YEARLY');
+
+      // RRULE:FREQ=DAILY;UNTIL=20210605;INTERVAL=1;BYDAY=WE;BYWEEKNO=3,9,11,13,15,17,19,21,41,43,45,47,49,51
+
+      _gCalEvent.recurrence.add(newRruleString);
+    }
+
+    return _gCalEvent;
+  }
+
+  Future<void> exportToGoogleCalendar(TimetablePage widget) async {
+    final Stream<List<UniEvent>> eventsStream = _events;
+    final List<UniEvent> streamElement = await eventsStream.first;
+
+    final List<g_cal.Event> _gCalEvents = [];
+
+    for (final UniEvent eventInstance in streamElement) {
+      //print('found event : ${eventInstance.classHeader.acronym}');
+      //print('which is a ${eventInstance.type}');
+      final g_cal.Event _gCalEvent = convertEvent(eventInstance);
+      _gCalEvents.add(_gCalEvent);
+    }
+
+    ///RRULE:FREQ=WEEKLY;UNTIL=20210605 works
+    ///RRULE:FREQ=WEEKLY;UNTIL=20210605T000000 doesn't work
+    ///RRULE:FREQ=WEEKLY;UNTIL=20210605T000000;INTERVAL=1;BYDAY=WE doesn't work
+
+    await insertGoogleEvents(_gCalEvents);
+  }
+
+  void prompt(String url) async {
+    print('Please go to the following URL and grant access:');
+    print('  => $url');
+    print('');
+
+    if (await canLaunch(url)) {
+      await launch(url);
+    } else {
+      print('Could not launch $url');
+    }
+  }
+
+  Future<void> insertGoogleEvents(List<g_cal.Event> _gCalEvents) async {
+    await clientViaUserConsent(
+            GoogleApiHelper.credentials, GoogleApiHelper.scopes, prompt)
+        .then(
+      //(AuthClient client) async {
+      (AutoRefreshingAuthClient client) async {
+        // TODO(bogpie): Remember if a user already gave access to his Google Calendar
+        // TODO(bogpie): Automatically close browser
+
+        final g_cal.CalendarApi calendarApi = g_cal.CalendarApi(client);
+        final g_cal.Calendar calendar = g_cal.Calendar();
+        g_cal.CalendarList calendarListNonIterable;
+        try {
+          calendarListNonIterable = await calendarApi.calendarList.list();
+        } catch (e) {
+          print('Error $e in getting calendars as a list');
+          return;
+        }
+        final List<g_cal.CalendarListEntry> calendarList =
+            calendarListNonIterable.items;
+        for (final g_cal.CalendarListEntry calendar in calendarList) {
+          if (calendar.summary == 'ACS UPB Mobile') {
+            try {
+              await calendarApi.calendars.delete(calendar.id);
+            } catch (e) {
+              print('Error $e in deleting calendar');
+            }
+            break;
+          }
+        }
+        // ignore: cascade_invocations
+        calendar
+          ..timeZone = 'Europe/Bucharest'
+          ..summary = 'ACS UPB Mobile'
+          ..description = 'Timetable imported from ACS UPB Mobile';
+
+        final g_cal.Calendar returnedCalendar =
+            await calendarApi.calendars.insert(calendar);
+
+        if (returnedCalendar is g_cal.Calendar) {
```

Comment:
Isn't this always true?

Reply:
It can complete with a [commons.ApiRequestError] if the API endpoint returned an error